### PR TITLE
Avoid transforming empty geometry in LinePolygonHighlight

### PR DESCRIPTION
**Pull request by nirvn**
@m-kuhn , I think this might fix / help avoid the crash you were witnessing yesterday.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d2cf128bfe990eea801a27032b35ddf89cc4f482:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-4c40573d515f5e5c0b5627c1ed1d71a9f65c9536-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-4c40573d515f5e5c0b5627c1ed1d71a9f65c9536-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-4c40573d515f5e5c0b5627c1ed1d71a9f65c9536-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-4c40573d515f5e5c0b5627c1ed1d71a9f65c9536-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for bcce380b01eadd4437874c364df45a1cd2ca121a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-d29006d3f637212472f7b837e6aad060c0221ff3-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-d29006d3f637212472f7b837e6aad060c0221ff3-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-d29006d3f637212472f7b837e6aad060c0221ff3-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1722-d29006d3f637212472f7b837e6aad060c0221ff3-x86.apk)

File: src/core/linepolygonhighlight.cpp [hunk 1]
```diff
@@ -37,18 +37,13 @@ QSGNode *LinePolygonHighlight::updatePaintNode( QSGNode *n, QQuickItem::UpdatePa
...
+      Q_ASSERT( geometry.type() != QgsWkbTypes::PointGeometry );
 
-      geometry = QgsGeometry( mGeometry->qgsGeometry() );
-
-      if ( mMapSettings )
-      {
-        QgsCoordinateTransform ct( mGeometry->crs(), mMapSettings->destinationCrs(), QgsProject::instance()->transformContext() );
-        geometry.transform( ct );
-      }
+      QgsCoordinateTransform ct( mGeometry->crs(), mMapSettings->destinationCrs(), QgsProject::instance()->transformContext() );
```
Comment (m-kuhn) [inline src/core/linepolygonhighlight.cpp:45]: Can we safely skip the mapsettings nullptr check?

Comment (nirvn) [reply src/core/linepolygonhighlight.cpp:45]: Yes, there's one at the very top of this function.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**